### PR TITLE
feat: add lua wireshark for pulsar

### LINKED ISSUE #13517: provide a lua  plugin for pulsar in  wireshark
**Is your feature request related to a problem? Please describe.**

https://github.com/apache/pulsar/tree/master/pulsar-client-cpp/wireshark

this is difficult in many different operation systems， I can write a Lua plugin for pulsar。 which will be easy to change and run in different systems， if it is ok， you can assigin to me

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by leizhiyuan**
Fixes #13517

### Motivation

add a Lua plugin

### Modifications

*Describe the modifications you've done.*

### Verifying this change
maybe need a code review

image

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

Check the box below or label this PR directly (if you have committer privilege).

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [ ] `no-need-doc` 
  
  (Please explain why)
  
- [x] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: wireshark/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+Edit->Preferences->Protocols->ProtoBuf->Protobuf search paths，add the path of PulsarApi.proto
+
+## add pulsar.lua to plugins
+
+open wireshark, go to 'About Wireshark', click  'Folders', then click 'Personal Lua Plugins'
+go to the plugin path, add put pulsar.lua to this path.
+
+## start to use
+
+this plugin will register a pulsar protocol automatically in 6650, you can use it to decode pulsar message now.
```
Comment (Anonymitaet) [inline wireshark/README.md:14]: @Demogorgon314 can you review the doc from the tech perspective? Thanks

**Review by Shoothzj — APPROVED** [review]

**Review by Demogorgon314 — APPROVED** [review]

File: wireshark/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# how to use 
+
+## prepare PulsarApi.proto file
+you need to put PulsarApi.proto to a separate path, open your wireshark,go to 
+Edit->Preferences->Protocols->ProtoBuf->Protobuf search paths，add the path of PulsarApi.proto
+
+## add pulsar.lua to plugins
+
+open wireshark, go to 'About Wireshark', click  'Folders', then click 'Personal Lua Plugins'
+go to the plugin path, add put pulsar.lua to this path.
```
Comment (Anonymitaet) [inline wireshark/README.md:10]:
```suggestion
1. Open Wireshark.

2. Go to **About Wireshark > Folders > Personal Lua Plugins > Plugin Path**.

3. Add pulsar.lua to this path.
```

do you mean this?

File: wireshark/README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+## start to use
```
Comment (Anonymitaet) [inline wireshark/README.md:12]:
```suggestion
## Step 3: start to use
```

File: wireshark/README.md (hunk 1 above)
Comment (Anonymitaet) [inline wireshark/README.md:14]:
```suggestion
This plugin registers a Pulsar protocol automatically in 6650. You can use it to decode Pulsar messages now.
```
English syntax details: https://docs.google.com/document/d/1lc5j4RtuLIzlEYCBo97AC8-U_3Erzs_lxpkDuseU0n4/edit#bookmark=id.e8uqh1awkcnp

**Review by Anonymitaet — CHANGES_REQUESTED** [review]

**Comment by BewareMyPower** [comment]
I think you should add the license in the LUA script.

**Review by Anonymitaet — APPROVED** [review]
Approve from the tech writing pespective

**Comment by BewareMyPower** [comment]
> Missing header in: /Users/runner/work/pulsar/pulsar/wireshark/pulsar.lua

The license is still missed. I think LUA is not commented by `/* ... */`. See https://en.wikibooks.org/wiki/Lua_Programming/comment. Before pushing your commit, you can run `mvn -B clean install -DskipTests` in your local env to ensure the license has been added correctly.

**Comment by leizhiyuan** [comment]
> > Missing header in: /Users/runner/work/pulsar/pulsar/wireshark/pulsar.lua
> 
> The license is still missed. I think LUA is not commented by `/* ... */`. See https://en.wikibooks.org/wiki/Lua_Programming/comment. Before pushing your commit, you can run `mvn -B clean install -DskipTests` in your local env to ensure the license has been added correctly.

image

it seems lua comment can not be recognized, could I ignore .lua in config?

**Comment by leizhiyuan** [comment]
see https://mycila.carbou.me/license-maven-plugin/

**Comment by Anonymitaet** [comment]
Hi @BewareMyPower does the author resolve your concern?

**Review by BewareMyPower — APPROVED** [review]

**Pull request merged**